### DISCUSSION TITLE
Why using "clsx" ?

### DISCUSSION THREAD

**Discussion by AbdullahSoomroOfficial** [shadcn-ui/ui#4668] [Q&A]
tailwind-merge uses twJoin to join class strings together, ensuring there are no conflicting classes. This means that twMerge already handles class merging efficiently by utilizing twJoin.

Given this, why would we use twMerge(clsx(<data>)), which essentially performs the class joining task twice? Wouldn't using clsx alongside twMerge be redundant?

[twJoin has the same performance characteristics as other class joining libraries like clsx](https://github.com/dcastil/tailwind-merge/blob/v2.5.2/docs/when-and-how-to-use-it.md#how-to-use-it)

**Comment by QuentinFrc** [comment]
As mentioned in tailwind-merge doc [here](https://github.com/dcastil/tailwind-merge/blob/v2.5.2/docs/api-reference.md#twjoin), `twJoin` doesn't support object arguments ([for this reason](https://github.com/dcastil/tailwind-merge/discussions/137#discussioncomment-3481605)).

So `clsx` is only here to handle object arguments in `cn` function.

> If you never pass objects in `cn(...args)`, for sure you can remove `clsx`

**Reply by AbdullahSoomroOfficial** [reply]
Thank you @QuentinFrc for the guidance. 🙏
